hikari/api/cache.py

```diff
@@ -1584,3 +1585,87 @@ def update_voice_state(
         builtins.NotImplementedError
             When called on a stateless cache implementation.
         """
+
+    @abc.abstractmethod
+    def delete_message(self, message_id: snowflakes.Snowflake) -> None:
+        """Remove a message object from the cache.
+
+        Parameters
+        ----------
+        message_id : snowflakes.Snowflake
+            The ID of the messages to remove the cache.
+
+        Raises
+        ------
+        builtins.NotImplementedError
+            When called on a stateless cache implementation.
+        """
+
+    @abc.abstractmethod
+    def delete_messages(self, message_ids: typing.AbstractSet[snowflakes.Snowflake]) -> None:
+        """Remove a messages object from the cache.
+
+        Parameters
+        ----------
+        message_ids : typing.Sequence[snowflakes.Snowflake]
+            The IDs of the messages to remove the cache.
+
+        Raises
+        ------
+        builtins.NotImplementedError
+            When called on a stateless cache implementation.
+        """
+
+    @abc.abstractmethod
+    def get_message(self, message_id: snowflakes.Snowflake) -> typing.Optional[messages.PartialMessage]:
+        """Get a message object from the cache.
+
+        Parameters
+        ----------
+        message_id : hikari.snowflakes.Snowflake
+            The ID of the message to get from the cache.
+
+        Returns
+        -------
+        typing.Optional[messages.PartialMessage]
+            The object of the message found in the cache or `builtins.None`.
+        """
+
+    @abc.abstractmethod
+    def set_message(self, message: messages.PartialMessage) -> None:
+        """Add a message object to the cache.
+
+        Parameters
+        ----------
+        message : messages.PartialMessage
+            The object of the message to add to the cache.
+
+        Raises
+        ------
+        builtins.NotImplementedError
+            When called on a stateless cache implementation.
+        """
+
+    @abc.abstractmethod
+    def update_message(
+        self, message: messages.PartialMessage
+    ) -> typing.Tuple[typing.Optional[messages.PartialMessage], typing.Optional[messages.PartialMessage]]:
+        """Update a message in the cache.
+
+        Parameters
+        ----------
+        message : messages.PartialMessage
+            The object of the message to update in the cache.
+
+        Returns
+        -------
+        typing.Tuple[typing.Optional[messages.PartialMessage], typing.Optional[messages.PartialMessage]]
+            A tuple of the old cached message object if found (else `builtins.None`
+            and the new cached message object if it could be cached (else
+            `builtins.None`).
```

Comment:
```suggestion
        typing.Tuple[typing.Optional[messages.PartialMessage], typing.Optional[messages.PartialMessage]]
            A tuple of the old cached message object if found (else `builtins.None`)
            and the new cached message object if it could be cached (else
            `builtins.None`).
```